code/datums/status_effects/xeno_buffs.dm

```diff
@@ -142,10 +141,25 @@
 ///Handles the link breaking due to distance
 /datum/status_effect/xeno_psychic_link/proc/handle_dist(datum/source)
 	SIGNAL_HANDLER
-	if(get_dist(owner, target_mob) > link_range)
-		to_chat(target_mob, span_xenowarning("You are too far away from [owner]."))
-		to_chat(owner, span_xenowarning("[target_mob] is too far away."))
-		owner.remove_status_effect(STATUS_EFFECT_XENO_PSYCHIC_LINK)
+	var/within_range_current = get_dist(owner, target_mob) <= link_range
+	if(within_range_current == within_range)
```

Comment:
i don't understand what you are trying to do

within_range_current, as you defined it, is a boolean (true of false)

You never check two booleans with ==